Sources/StreamChat/Workers/Background/AttachmentQueueUploader.swift

```diff
@@ -59,46 +59,44 @@ class AttachmentQueueUploader: Worker {
     private func handleChanges(changes: [ListChange<AttachmentDTO>]) {
         guard !changes.isEmpty else { return }
 
+        // Only start uploading attachment when inserted and it is present in pendingAttachmentIds
         database.backgroundReadOnlyContext.perform { [weak self] in
-            var wasEmpty: Bool = false
             self?._pendingAttachmentIDs.mutate { pendingAttachmentIDs in
-                wasEmpty = pendingAttachmentIDs.isEmpty
-                changes.pendingUploadAttachmentIDs.forEach {
+                let newAttachmentIds = Set(changes.attachmentIDs).subtracting(pendingAttachmentIDs)
+                newAttachmentIds.forEach {
                     pendingAttachmentIDs.insert($0)
                 }
```

Comment:
Alternative could be `pendingAttachmentIDs.formUnion(newAttachmentIds)`. But it does not matter much.